src/Compiler/Service/ServiceAnalysis.fs

```diff
@@ -451,3 +452,1091 @@ module UnusedDeclarations =
+        | AmpAmp
+
+        /// :?>
+        | Downcast
+
+        /// :>
+        | Upcast
+
+        /// =…
+        | Eq
+
+        /// |…
+        | Bar
+
+        /// &…
+        | Amp
+
+        /// $…
+        | Dollar
+
+        /// >…
+        | Greater
+
+        /// <…
+        | Less
+
+        /// !=…
+        | BangEq
+
+        /// ^…
+        | Hat
+
+        /// @…
+        | At
+
+        /// ::
+        | Cons
+
+        /// :?
+        | TypeTest
+
+        /// -…
+        | Sub
+
+        /// +…
+        | Add
+
+        /// %…
+        | Mod
+
+        /// /…
+        | Div
+
+        /// *…
+        | Mul
+
+        /// **…
+        | Exp
+
+        /// - x
+        | UnaryPrefix
+
+        /// f x
+        | Apply
+
+        /// -x, !… x, ~~… x
+        | High
+
+        // x.y
+        | Dot
+
+    module Precedence =
+        let sameKind prec1 prec2 = prec1 = prec2
+
+        let compare prec1 prec2 =
+            match prec1, prec2 with
+            | Dot, Dot -> 0
+            | Dot, _ -> 1
+            | _, Dot -> -1
+
+            | High, High -> 0
+            | High, _ -> 1
+            | _, High -> -1
+
+            | Apply, Apply -> 0
+            | Apply, _ -> 1
+            | _, Apply -> -1
+
+            | UnaryPrefix, UnaryPrefix -> 0
+            | UnaryPrefix, _ -> 1
+            | _, UnaryPrefix -> -1
+
+            | Exp, Exp -> 0
+            | Exp, _ -> 1
+            | _, Exp -> -1
+
+            | (Mod | Div | Mul), (Mod | Div | Mul) -> 0
+            | (Mod | Div | Mul), _ -> 1
+            | _, (Mod | Div | Mul) -> -1
+
+            | (Sub | Add), (Sub | Add) -> 0
+            | (Sub | Add), _ -> 1
+            | _, (Sub | Add) -> -1
+
+            | TypeTest, TypeTest -> 0
+            | TypeTest, _ -> 1
+            | _, TypeTest -> -1
+
+            | Cons, Cons -> 0
+            | Cons, _ -> 1
+            | _, Cons -> -1
+
+            | (Hat | At), (Hat | At) -> 0
+            | (Hat | At), _ -> 1
+            | _, (Hat | At) -> -1
+
+            | (Eq | Bar | Amp | Dollar | Greater | Less | BangEq), (Eq | Bar | Amp | Dollar | Greater | Less | BangEq) -> 0
+            | (Eq | Bar | Amp | Dollar | Greater | Less | BangEq), _ -> 1
+            | _, (Eq | Bar | Amp | Dollar | Greater | Less | BangEq) -> -1
+
+            | (Downcast | Upcast), (Downcast | Upcast) -> 0
+            | (Downcast | Upcast), _ -> 1
+            | _, (Downcast | Upcast) -> -1
+
+            | AmpAmp, AmpAmp -> 0
+            | AmpAmp, _ -> 1
+            | _, AmpAmp -> -1
+
+            | BarBar, BarBar -> 0
+            | BarBar, _ -> 1
+            | _, BarBar -> -1
+
+            | Comma, Comma -> 0
+            | Comma, _ -> 1
+            | _, Comma -> -1
+
+            | ColonEquals, ColonEquals -> 0
+            | ColonEquals, _ -> 1
+            | _, ColonEquals -> -1
+
+            | Set, Set -> 0
+
+    /// Associativity/association.
+    type Assoc =
```

Comment:
BTW I cannot help having the IngSoc association here, partially because you have 2 + 2 = 5 in the PR description :D